docs/backend/relations.md

```diff
@@ -0,0 +1,586 @@
+from plone.autoform.interfaces import IFormFieldProvider
+from plone.supermodel import model
+from z3c.relationfield.schema import RelationChoice
+from z3c.relationfield.schema import RelationList
+from zope.interface import provider
+
+
+def minion_favorites(context):
+    portal = api.portal.get()
+    minions_path = "/".join(portal["minions"].getPhysicalPath())
+    one_eyed_minions_path = "/".join(portal["one-eyed-minions"].getPhysicalPath())
+    return [
+            {
+                "title": "Current Content",
+                "path": "/".join(context.getPhysicalPath())
+            }, {
+                "title": "Minions",
+                "path": minions_path,
+            }, {
+                "title": "One eyed minions",
+                "path": one_eyed_minions_path,
+            }
+        ]
+
+
+@provider(IFormFieldProvider)
+class IHaveMinions(model.Schema):
+
+    minions = RelationList(
+        title="My minions",
+        default=[],
+        value_type=RelationChoice(
+            source=CatalogSource(
+                portal_type=["one_eyed_minion", "minion"],
+                review_state="published",
+            )
+        ),
+        required=False,
+    )
+    directives.widget(
+        "minions",
+        RelatedItemsFieldWidget,
+        pattern_options={
+            "mode": "auto",
+            "favorites": minion_favorites,
+            }
+        )
+```
+
+## RelationFields through the web or in xml
+
+It is surprisingly easy to create RelationFields through the web
+
+- Using the Dexterity schema editor, add a new field and select _Relation List_ or _Relation Choice_, depending on whether you want to relate to multiple items or not.
+- When configuring the field you can even select the content type the relation should be limited to.
+
+When you click on `Edit XML field model` you will see the fields in the XML schema:
+
+RelationChoice:
+
+```{code-block} python
+:linenos:
+
+<field name="boss" type="z3c.relationfield.schema.RelationChoice">
+  <description/>
+  <required>False</required>
+  <title>Boss</title>
+</field>
+```
+
+RelationList:
+
+```{code-block} python
+:linenos:
+
+<field name="underlings" type="z3c.relationfield.schema.RelationList">
+  <description/>
+  <required>False</required>
+  <title>Underlings</title>
+  <value_type type="z3c.relationfield.schema.RelationChoice">
+    <title i18n:translate="">Relation Choice</title>
+    <portal_type>
+      <element>Document</element>
+      <element>News Item</element>
+    </portal_type>
+  </value_type>
+</field>
+```
+
+## Using different widgets for relations
+
+```{todo}
+Support for `StaticCatalogVocabulary` in Volto is currently still missing.
+```
+
+Often the standard widget for relations is not what you want since it can be hard to navigate to the content you want to relate to.
+
+If you want to use checkboxes, radiobuttons or a selection-dropdown you need to use `StaticCatalogVocabulary` instead of `CatalogSource` to specify your options.
```

Comment:
This is some how astonishing to motivate a switch to new `StaticCatalogVocabulary` with display decision. Why not start this chapter with `StaticCatalogVocabulary`. Later the traditional way can be mentioned, to understand existing code.

Reply:
Sounds like a plan. 

> From time to time I think it would be good to stop defining target sets of a relation field in its widget. Cause this is a constraint of its data structure (Plone: schema) and not a matter of the display (Plone: widget).

I guess you mean `baseCriteria`. I don't really like it but it optimizes the catalog-query. Feel free to remove that since it is a rather advanced hack.

Reply:
`StaticCatalogVocabulary` only works with widgets other than `RelatedItemsFieldWidget`. So the default is still that and only if you require other widgets you can use `StaticCatalogVocabulary`.